3rdparty/google_cloud_texttospeech/bin/text2wave

```diff
@@ -1,11 +1,82 @@
 #!/usr/bin/env python
 
 import argparse
+from distutils.version import LooseVersion
+import hashlib
+import os
+import shutil
+import sys
 
-from google.cloud import texttospeech
+from google.cloud.texttospeech import TextToSpeechClient
+import pkg_resources
 import rospy
 
 
+# The submodules enums and types have been removed from v2.0.0
+# https://github.com/googleapis/python-texttospeech/blob/master/UPGRADING.md#enums-and-types  # NOQA
+texttospeech_version = pkg_resources.get_distribution(
+    "google-cloud-texttospeech").version
+if LooseVersion(texttospeech_version) >= LooseVersion('2.0.0'):
+    from google.cloud.texttospeech import AudioConfig
+    from google.cloud.texttospeech import AudioEncoding
+    from google.cloud.texttospeech import SsmlVoiceGender
+    from google.cloud.texttospeech import SynthesisInput
+    from google.cloud.texttospeech import VoiceSelectionParams
+else:
+    import google.cloud.texttospeech
+    AudioEncoding = google.cloud.texttospeech.enums.AudioEncoding
+    SsmlVoiceGender = google.cloud.texttospeech.enums.SsmlVoiceGender
+    AudioConfig = google.cloud.texttospeech.types.AudioConfig
+    SynthesisInput = google.cloud.texttospeech.types.SynthesisInput
+    VoiceSelectionParams = google.cloud.texttospeech.types.VoiceSelectionParams
+
+cache_enabled = os.environ.get(
+    'GOOGLE_CLOUD_TEXTTOSPEECH_CACHE_ENABLED', False)
```

Comment:
This could be `param`.
PATH, PYTHONENCODING, GOOGLE_APPLICATION_CREDENTIALS must be env, because that is used in google.cloud.texttospeech modules, but GOOGLE_CLOUD_TEXTTOSPEECH_CACHE_ENABLED is only used in this file.

I also feel set cache enable could be True, by default.

Reply:
I found the solution has a problem.
`text2wave` will be called at sound_play node using `os.system` https://github.com/ros-drivers/audio_common/blob/master/sound_play/scripts/soundplay_node.py#L267 .
I want to specify namespace to pass as ROS param, but I can't specify it correctly just by calling it in `os.system`.

Is there a way to correctly give the ROS namespace to the `text2wave` called in `os.system`?

Reply:
I see, then, using env variable make sense. I think using env is ok.

Reply:
OK.

Also, I modified the code for caching to default behavior.